aten/src/ATen/native/TensorTransformations.cpp

```diff
@@ -9,51 +10,90 @@
 namespace at {
 namespace native {
 
+template <typename scalar_t>
+void inline flip_cpu_kernel(
+  const int64_t numel,
+  const int64_t total_dims,
+  const int64_t flip_dims_size,
+  const int64_t* stride_contiguous_d,
+  const int64_t* flip_dims_d,
+  const int64_t* strides_d,
+  const int64_t* sizes_d,
+  const scalar_t* in_tensor_d,
+  scalar_t* out_tensor_d
+){
+  int64_t i;
+
+  #pragma omp parallel for private(i) if (numel > 1000)
+  for (i = 0; i < numel; i++) {
+    int64_t cur_indices = i;
+    int64_t rem = 0;
+    int64_t dst_offset = 0;
+
+    for (int64_t d = 0; d < total_dims; d++) {
+      int64_t temp = cur_indices;
+      cur_indices = cur_indices / stride_contiguous_d[d];
+      rem = temp - cur_indices * stride_contiguous_d[d];
+      for (int64_t j = 0; j < flip_dims_size; j++) {
+        if (d == flip_dims_d[j]) cur_indices = sizes_d[d] - 1 - cur_indices;
+      }
+      dst_offset += cur_indices * strides_d[d];
+      cur_indices = rem;
+    }
+    out_tensor_d[i] = in_tensor_d[dst_offset];
+  }
+}
+
 Tensor flip_cpu(const Tensor& self, IntList dims) {
-  const int64_t total_dims = self.dim(), flip_dims_size = dims.size();
-  flip_check_errors(total_dims, flip_dims_size, dims);
+  auto in_tensor = self;
+  const int64_t total_dims = self.dim();
 
+  dim_list_to_bitset(dims, total_dims);
```

Comment:
This call seems unused?

Reply:
It seems to me that if you pass over the bitset to the kernel above, it will be much faster.

Reply:
yeah! you're right!

Reply:
still not using bitset in the changes, the thing is I will need to get `flip_dims` at the kernel, although I can still get the same thing from a bitset by going through the entire set, but it will be a longer loop, and thus the hesitation to use it here